nucliadb_node/nucliadb_node/pull.py

```diff
@@ -307,10 +307,13 @@ async def subscription_worker(self, msg: Msg):
             msg, nats_consumer_settings.nats_ack_wait * 0.66
         ), sm.lock:
             try:
+                status = None
                 if pb.typemessage == TypeMessage.CREATION:
-                    await self.set_resource(pb)
+                    status = await self.set_resource(pb)
                 elif pb.typemessage == TypeMessage.DELETION:
-                    await self.delete_resource(pb)
+                    status = await self.delete_resource(pb)
+                if status is not None and status.status != OpStatus.OK:
+                    raise Exception(status.detail)
```

Comment:
Instead of a generic exception, I think it'd be better to create a custom for instance `NodeIndexOpError` or something like that